src/edge_proxy/cache.py

```diff
@@ -57,11 +57,12 @@ def _put_environment(
         environment_document: dict[str, Any],
     ) -> None:
         self._environment_cache[environment_api_key] = environment_document
-        for identity_document in environment_document.get("identity_overrides") or []:
-            if identifier := identity_document.get("identifier"):
-                self._identity_override_cache[environment_api_key][identifier] = (
-                    identity_document
-                )
+        self._identity_override_cache[environment_api_key] = {
+            identifier: identity_document
+            for identity_document in environment_document.get("identity_overrides")
+            or []
+            if (identifier := identity_document.get("identifier"))
+        }
```

Comment:
Just for cognitive load reading this comprehension, I'd rather do something like: 

```suggestion
        new_overrides = environment_document.get("identity_overrides") or []
        self._identity_override_cache[environment_api_key] = {
            identifier: identity_document
            for identity_document in new_overrides
            if (identifier := identity_document.get("identifier"))
        }
```

or even just:

```suggestion
if new_overrides := environment_document.get("identity_overrides"):
    self._identity_override_cache[environment_api_key] = {
        identifier: identity_document
        for identity_document in new_overrides
        if (identifier := identity_document.get("identifier"))
    }
```

I guess they both amount to the same at the end of the day.